ci/setup-tests.ps1

```diff
@@ -1,23 +1,72 @@
 # Expects gdk_home
 param(
     [string] $build_output_dir,
-    [string] $project_path,
-    [string] $unreal_path = "$((Get-Item `"$($PSScriptRoot)`").parent.parent.FullName)\UnrealEngine" ## This should ultimately resolve to "C:\b\<number>\UnrealEngine".
+    [string] $unreal_path = "$((Get-Item `"$($PSScriptRoot)`").parent.parent.FullName)\UnrealEngine", ## This should ultimately resolve to "C:\b\<number>\UnrealEngine".
+    [string] $test_repo_branch,
+    [string] $test_repo_url,
+    [string] $test_repo_map,
+    [string] $test_repo_uproject_path,
+    [string] $test_repo_path,
+    [string] $msbuild_exe
 )
 
 # Copy the built files back into the SpatialGDK folder, to have a complete plugin
 # The trailing \ on the destination path is important!
 Copy-Item -Path "$build_output_dir\*" -Destination "$gdk_home\SpatialGDK\" -Recurse -Container -ErrorAction SilentlyContinue
 
+# Update spatial to newest version
+Start-Process spatial "update" -Wait -ErrorAction Stop -NoNewWindow
+
+# Clean up testing project (symlinks could be invalid during initial cleanup - leaving the project as a result)
+if (Test-Path $test_repo_path) {
+    Write-Log "Removing existing project"
+    Remove-Item $test_repo_path -Recurse -Force
+    if (-Not $?) {
+        Throw "Failed to remove existing project at $($test_repo_path)."
+    }
+}
+
+# Clone and build the testing project
+Write-Log "Downloading the testing project from $($test_repo_url)"
+Git clone -b $test_repo_branch $test_repo_url $test_repo_path
```

Comment:
If you want a very small optimisation, you could add --depth 1 (I think, refer to ExampleProject CI), to reduce the size of the download

Reply:
Good one! Added it now